v22.1/query-behavior-troubleshooting.md

```diff
@@ -54,105 +56,202 @@ You can look more closely at the behavior of a statement by visualizing a statem
+    In the following example, the trace shows that there is significant latency between a push attempt on a transaction that is holding a [lock](architecture/transaction-layer.html#writing) (56.85ms) and that transaction being committed (131.37ms).
 
     <img src="{{ 'images/v22.1/jaeger-trace-transaction-contention.png' | relative_url }}" alt="Jaeger Trace Log Messages" style="border:1px solid #eee;max-width:100%" />
 
-## `SELECT` statement performance issues
+<a id="query-is-always-slow"></a>
+
+### Queries are always slow
+
+If you have consistently slow queries in your cluster, use the [Statement Details](ui-statements-page.html#statement-details-page) page to drill down to an individual statement and [collect diagnostics](ui-statements-page.html#diagnostics) for the statement. A diagnostics bundle contains a record of transaction events across nodes for the SQL statement.
+
+You can also use an [`EXPLAIN ANALYZE`](explain-analyze.html) statement, which executes a SQL query and returns a physical query plan with execution statistics. You can use query plans to troubleshoot slow queries by indicating where time is being spent, how long a processor (i.e., a component that takes streams of input rows and processes them according to a specification) is not doing work, etc.
+
+Cockroach Labs recommends sending either the diagnostics bundle (preferred) or the `EXPLAIN ANALYZE` output to our [support team](support-resources.html) for analysis.
+
+### Queries are sometimes slow
+
+If the query performance is irregular:
+
+1.  Run [`SHOW TRACE`](show-trace.html) for the query twice: once when the query is performing as expected and once when the query is slow.
+
+2.  [Contact support](support-resources.html) to help analyze the outputs of the `SHOW TRACE` command.
+
+### `SELECT` statements are slow
 
 The common reasons for a sub-optimal `SELECT` performance are inefficient scans, full scans, and incorrect use of indexes. To improve the performance of `SELECT` statements, refer to the following documents:
 
 -  [Table scan best practices](performance-best-practices-overview.html#table-scan-best-practices)
 
 -  [Indexes best practices](schema-design-indexes.html#best-practices)
 
-## Query is always slow
+### `SELECT` statements with `GROUP BY` columns are slow
 
-If you have consistently slow queries in your cluster, use the [Statement Details](ui-statements-page.html#statement-details-page) page to drill down to an individual statement and [collect diagnostics](ui-statements-page.html#diagnostics) for the statement. A diagnostics bundle contains a record of transaction events across nodes for the SQL statement.
+Suppose you have a slow query has a `GROUP BY` clause, uses an index that has a `STORING` clause, and some or all of the columns in the query's `GROUP BY` clause are part of the index's `STORING` clause and are not index key columns. For example:
```

Comment:
Suppose you have a slow query that

-  Has a `GROUP BY` clause.
-  Uses an index that has a `STORING` clause.
-  Where some or all of the columns in the query's `GROUP BY` clause are part of the index's `STORING` clause and are **not** index key columns. 

For example: